portfolio/src/main/webapp/script.js

```diff
@@ -61,3 +61,171 @@ function addRandomFunFact() {
+ * If 'active' is true, then add 'active' to its className.
+ */
+function createPageElement(text, active) {
+  const pageElement = document.createElement('li');
+  pageElement.className = 'page-item';
+  if (active) { pageElement.classList.add('active'); }
+  
+  const linkElement = document.createElement('a');
+  linkElement.className = 'page-link';
+  linkElement.innerText = text;
+  pageElement.appendChild(linkElement);
+
+  pageElement.addEventListener('click', event => changePage(event));
+  return pageElement;
+}
+
+/* 
+ * When a page item is clicked, changes the active page element and displays the comments on that page.
+ */
+function changePage(event) {
+  const currentPage = document.querySelector('.page-item.active');
+  const pageText = event.currentTarget.firstElementChild.innerText;
+  
+  if (pageText == 'Previous') {
+    // If the currentPage is the first page, then return without change.
+    if (currentPage.firstElementChild.innerText == 1) { return; }
+
+    currentPage.previousSibling.classList.add('active');
+  } else if (pageText == 'Next') {
+    // If the currentPage is the last page, then return without change.
+    const numPages = document.querySelectorAll('#page-list li').length;
+    if (currentPage.firstElementChild.innerText == numPages - 2) { return; }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetches comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/* 
+ * Fetches the comments data on the current page from the server and displays them.
+ */
+function fetchComments(createNewPagination = false) {
+  const currentPage = document.querySelector('.page-item.active');
+  var currentPageNumber = currentPage.firstElementChild.innerText;
+
+  const selectElement = document.getElementById('num-comments-per-page');
+  const commentsPerPage = selectElement.options[selectElement.selectedIndex].value;
+
+  fetch('/data').then(response => response.json()).then(comments => {
+    const commentsListElement = document.getElementById('comments-list');
+    commentsListElement.innerHTML = '';
+
+    if (createNewPagination) {
+      currentPageNumber = createPagination(Math.ceil(comments.length / commentsPerPage), currentPageNumber);
+    }
+
+    const startIndex = (currentPageNumber - 1) * commentsPerPage;
+    const endIndex = Math.min(Number(commentsPerPage) + Number(startIndex), comments.length);
+    for (var i = startIndex; i < endIndex; i++) {
+      commentsListElement.appendChild(createCommentElement(comments[i]));
+    }
+  });
+}
+
+/* 
+ * Creates an <li> element containing a comment.
+ */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const divElement = document.createElement('div');
```

Comment:
rename it to something like commentContainer?

Reply:
Ok